restaurants/src/app/components/restaurants/restaurants.component.ts

```diff
@@ -1,41 +1,43 @@
 import { Component, Input, OnInit } from "@angular/core";
 import { Restaurant } from "../../models/restaurants.model";
-import { RESTAURANTS } from "src/app/services/restaurants.service";
 import { throwIfEmpty } from "rxjs";
+import { v4 as uuidv4 } from 'uuid';
+import { RestaurantsService } from "src/app/services/restaurants.service";
 
 @Component({
     selector: 'app-restaurants',
     templateUrl: './restaurants.component.html',
     styleUrls: ['./restaurants.component.scss']
   })
-  export class RestaurantsComponent {
-    Restaurant = RESTAURANTS;
+  export class RestaurantsComponent implements OnInit {
+    constructor(private restaurantsService: RestaurantsService){}
+    restaurants: {[id: string]: Restaurant} = {};
+
     name: string = '' ;
     rank: number = 0;
     id: number = 0;
 
-    newRestaurant: Restaurant = {id: 0, name: '', rank: 0};
+    newRestaurant: Restaurant = { name: '', rank: 0};
```

Comment:
I don't see this variable being used, can it be deleted?

Reply:
yes